### PR TITLE
Add CodeQL workflow

### PR DESCRIPTION
**Pull request by sleberknight**
* De-lomboks before build
* Runs extended security and quality suites

### CODE REVIEW

**Comment by sleberknight** [comment]
Compilation failed because one class (`WebTargetHelper`) uses Lombok's `@Delegate`, which (for whatever reason) is not including the delegated methods in the de-lomboked code. This must be something wrong with how the [advanced-security/delombok](https://github.com/advanced-security/delombok) does the delombok, because running a regular delombok (e.g. in IntelliJ or the Lombok Maven plugin, or manually via `java -jar lombok.jar delombok ...` all produce the expected code. I've skimmed the code of the action and there's nothing obvious, e.g. it doesn't intentionally exclude experimental Lombok features like `@Delegate`.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - name: Delombok
+      uses: advanced-security/delombok@main
+
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v2
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v2
+      with:
+        category: "/language:${{matrix.language}}"
```
Comment (sleberknight) [inline .github/workflows/codeql.yml:58]: Add the newline

**Comment by sleberknight** [comment]
This is blocked until #859 is resolved. Then, rebase to this branch.

**Comment by sleberknight** [comment]
@dependabot rebase

**Review by terezivy — APPROVED** [review]

**Comment by sleberknight** [comment]
This PR branch is "16 commits ahead, 14 commits behind master" and since I don't want to deal with trying to fix all that, closing this and will create a new PR.

**Pull request closed without merging**